client/src/main/java/org/apache/rocketmq/client/trace/TraceDataEncoder.java

```diff
@@ -104,9 +104,8 @@
                     subAfterContext.setContextCode(Integer.parseInt(line[6]));
                 }
                 // compatible with the old version
-                if (line.length >= 9) {
+                if (line.length >= 8) {
                     subAfterContext.setTimeStamp(Long.parseLong(line[7]));
-                    subAfterContext.setGroupName(line[8]);
```

Comment:
The code here is used for compatible reason, if the message is produced by old versions of producer, there will be group name.

Reply:
Get it. 
I removed it because I thought this value is useless, maybe add the branch line.length >= 9 back is a better way. 
```
        // compatible with the old version
        if (line.length >= 8) {
            subAfterContext.setTimeStamp(Long.parseLong(line[7]));
        }
        if (line.length >= 9) {
            subAfterContext.setGroupName(Long.parseLong(line[8]));
        }
```